pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -220,10 +231,43 @@ func sortedKeys(m map[string]string) []string {
 	return keys
 }
 
-// getUserData returns the exact same string for equivalent input,
+func (p *LaunchTemplateProvider) getBottleRocketUserData(ctx context.Context, constraints *v1alpha1.Constraints, additionalLabels map[string]string) (string, error) {
+	var userData string
+	userData += fmt.Sprintf(`[settings.kubernetes]
+cluster-name = "%s"
+api-server = "%s"
+`,
+		injection.GetOptions(ctx).ClusterName,
+		injection.GetOptions(ctx).ClusterEndpoint)
+
+	if constraints.KubeletConfiguration.ClusterDNS != nil {
+		userData += fmt.Sprintf("cluster-dns-ip = \"%s\"", constraints.KubeletConfiguration.ClusterDNS)
+	}
+
+	caBundle, err := p.GetCABundle(ctx)
+	if err != nil {
+		return "", fmt.Errorf("getting ca bundle for user data, %w", err)
+	}
+	if caBundle != nil {
+		userData += fmt.Sprintf("cluster-certificate = \"%s\"\n", *caBundle)
+	}
+
+	nodeLabelArgs := functional.UnionStringMaps(additionalLabels, constraints.Labels)
+	if len(nodeLabelArgs) > 0 {
+		userData += `[settings.kubernetes.node-labels]
+`
+		for key, val := range nodeLabelArgs {
+			userData += fmt.Sprintf("\"%s\" = \"%s\"", key, val)
+		}
+	}
+
```

Comment:
What about seeting [taints](https://github.com/bottlerocket-os/bottlerocket#kubernetes-settings) here? To make sure we are  on par with what `getEKSOptimizedUserData` offers.

Reply:
Added a section for this.